py/compile.c

```diff
@@ -782,6 +782,32 @@ STATIC qstr compile_funcdef_helper(compiler_t *comp, mp_parse_node_struct_t *pns
     // compile the function definition
     compile_funcdef_lambdef(comp, fscope, pns->nodes[1], PN_typedargslist);
 
+    #if MICROPY_ENABLE_DOC_STRING
+    // look for the first statement
+    mp_parse_node_t pn = pns->nodes[3]; // the function body/suite
+    if (MP_PARSE_NODE_IS_STRUCT_KIND(pn, PN_suite_block_stmts)) {
```

Comment:
This check already seems to be done by `check_for_doc_string`?

i.e. this whole thing could be a call to `check_for_doc_string`, the only difference is how it emits the store of `__doc__` ?